### PR TITLE
Aveybranch

### PR DESCRIPTION
**Pull request by dspages**
Adds bonus damage for rear attacks and critical hits.

Currently, basic attacks inflict an average of 6 damage, plus one half of effective strength (with a random plus or minus 5).

Now, if you attack from behind, you do additional bonus damage which is an average of three plus one quarter your agility (with a random plus or minus 2).

Furthermore, there are rare critical hits that inflict an additional +50% damage on top of whatever other damage you do. The chance of these is 1% per ten points of effective intelligence, but the chance is doubled if you are attacking from behind or from the side.

### CODE REVIEW

File: Assets/Scripts/CreatureStats.cs [hunk 1]
```diff
@@ -133,19 +140,70 @@ private void DisplayPopup(string text)
         textLabelText = text;
     }
 
+    // Returns true if floats are within 15.0f of each other.
+    // compare Mathf.Approximately.
+    private bool VeryApproximately(float f1, float f2)
```
Comment (emshuttles) [inline Assets/Scripts/CreatureStats.cs]: I know Unity named their function Approximately, but that seems like a bad name to me. Could we call it AreApproximate() or something?

Comment (emshuttles) [inline Assets/Scripts/CreatureStats.cs]: I know Unity named their function Approximately, but that seems like a bad name to me. Could we call it AreApproximate() or something?

File: Assets/Scripts/CreatureStats.cs [hunk 2]
```diff
@@ -133,19 +140,70 @@ private void DisplayPopup(string text)
...
+        return false;
+    }
+
+    // Returns true if the target is being attacked from the side or rear.
+    // Side/read attacks double the chance of a critical hit.
+    public bool IsFlanking(CreatureStats target)
+    {
+        float rotation1 = target.transform.eulerAngles.y;
+        float rotation2 = transform.eulerAngles.y;
+        bool flanking = !VeryApproximately(Mathf.Abs(rotation1 - rotation2), 180.0f);
```
Comment (emshuttles) [inline Assets/Scripts/CreatureStats.cs]: Can we simplify this by writing it like so:
`return VeryApproximately(Mathf.Abs(rotation1 - rotation2), 0f);`

File: Assets/Scripts/CreatureStats.cs [hunk 3]
```diff
@@ -133,19 +140,70 @@ private void DisplayPopup(string text)
...
+        return flanking;
+    }
+
+    // Returns true if the target is being attacked from the rear.
+    // Rear attacks do a bonus 0-10 damage plus one half of agility.
+    public bool IsBehind(CreatureStats target)
```
Comment (emshuttles) [inline Assets/Scripts/CreatureStats.cs:161]: Is this a repeat of IsFlanking()?

File: Assets/Scripts/CreatureStats.cs [hunk 4]
```diff
@@ -133,19 +140,70 @@ private void DisplayPopup(string text)
...
-        target.DisplayPopup(dam + " damage inflicted!");
+        int dam = DamageInflicted();
+        if (IsBehind(target))
+        {
+            DisplayPopup("Rear attack!");
+            dam += BonusRearDamage();
+        }
+        if (IsCrit(target))
+        {
+            target.DisplayPopup("CRITICAL HIT! " + dam + " damage inflicted!");
```
Comment (emshuttles) [inline Assets/Scripts/CreatureStats.cs:198]: Should this go after you add another 50% to the damage?

File: Assets/Scripts/CreatureStats.cs (hunk 3 above)
Comment (dspages) [reply Assets/Scripts/CreatureStats.cs:161]: No - rear attacks include attacks from behind only. Flanking attacks include attacks from behind, left side, or right side.

File: Assets/Scripts/CreatureStats.cs (hunk 4 above)
Comment (dspages) [reply Assets/Scripts/CreatureStats.cs:198]: Yeah - in this case, stacking multipliers is the intended behavior. Though we can always revise it later if need-be.

File: Assets/Scripts/CreatureStats.cs (hunk 1 above)
Comment (dspages) [reply Assets/Scripts/CreatureStats.cs]: Renamed for clarity, thanks for the suggestion!

File: Assets/Scripts/CreatureStats.cs (hunk 2 above)
Comment (dspages) [reply Assets/Scripts/CreatureStats.cs]: Good idea. Done.

**Pull request merged**